### PR TITLE
 D2K - Add Harkonnen Mission 4

### PR DESCRIPTION
**Pull request by MustaphaTR**

### CODE REVIEW

**Comment by CH4Code** [comment]
Tested the map on easy and hard. On easy in original d2k it is sufficient to destroy one fremen sietch. Seems fine otherwise 👍

**Comment by MustaphaTR** [comment]
> On easy in original d2k it is sufficient to destroy one fremen sietch.

Are you sure about that, i don't think it is possible to code something like that in vanilla D2k, but i'll test.

**Comment by CH4Code** [comment]
Yes, im sure - just palyed it in original dune2k today. I just dont know 100% which difficulty I chose to play the campaign at, most likely easy or normal. Hard is realy tough..^^
The code should be easy... 
```
Trigger.OnKilledOrCaptured(sietch a, primaryfulfilled() )
Trigger.OnKilledOrCaptured(sietch b, primaryfulfilled() ) 
function primaryfulfilled( ) MissionObectives.MarkCompletedObjective(killFremen) end
```

**Comment by MustaphaTR** [comment]
I was talking about original's trigger system, i can code that on Lua. I don't remember anything about difficulty to be in there. I'm checking right now, couldn't find time yet.

**Comment by MustaphaTR** [comment]
That doesn't look correct (8 and 9):
resim

In original different fremen bases have different owners, bottom one is Emparor's actually. That trigger is missing a check for Emparor's Base and Units destroyed.

I'll consider this as a bug of original game and what happens in this PR (on OpenRA) looks more like a correct behaviour. So i'm not touching anything.

**Comment by CH4Code** [comment]
Well, then 👍 from me.

File: mods/d2k/maps/harkonnen-04/harkonnen04.lua [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+end
+
+InitObjectives = function()
+	Trigger.OnObjectiveAdded(player, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "New " .. string.lower(p.GetObjectiveType(id)) .. " objective")
+	end)
+
+	KillAtreides = player.AddPrimaryObjective("Destroy the Atreiedes.")
+	KillFremen = player.AddPrimaryObjective("Destroy the Fremen.")
+	KillHarkonnen = atreides.AddPrimaryObjective("Kill all Atreides units.")
```
Comment (ltem) [inline mods/d2k/maps/harkonnen-04/harkonnen04.lua]: Shouldn't this be "Kill all Harkonnen units."?

**Comment by ltem** [comment]
One question but otherwise :+1:

**Comment by MustaphaTR** [comment]
Updated.

**Comment by ltem** [comment]
:+1:

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/ee74ba4b8f96fc4eee822c3939cbf873d13624d3...244d86d05a79da2ce6204e5c40c6ddb53e191209

**Pull request merged**